stan/math/prim/prob/wiener5_lpdf.hpp

```diff
@@ -0,0 +1,428 @@
+                         msg_str.c_str());
+    }
+  }
+
+  if (!include_summand<propto, T_y, T_a, T_t0, T_w, T_v, T_sv>::value) {
+    return 0;
+  }
+
+  double lerror_bound_dens = log(1e-6);  // precision for density
+  double lerror_bound = log(prec);       // precision for derivatives
+  double labstol_wiener5 = log(1e-12);   // eps_abs(wiener5)
+  double dens = 0.0;
+  double ld = 0.0;
+  operands_and_partials<T_y_ref, T_a_ref, T_t0_ref, T_w_ref, T_v_ref, T_sv_ref>
+      ops_partials(y_ref, a_ref, t0_ref, w_ref, v_ref, sv_ref);
+
+  static constexpr double LOG_FOUR = LOG_TWO + LOG_TWO;
+
+  // calculate density and partials
+  for (size_t i = 0; i < N; i++) {
+    // Calculate 4-parameter model without inter-trial variabilities (if
+    // sv_vec[i] == 0) or 5-parameter model with inter-trial variability in
+    // drift rate (if sv_vec[i] != 0)
+    const double y_val = y_vec.val(i);
+    const double a_val = a_vec.val(i);
+    const double t0_val = t0_vec.val(i);
+    const double w_val = w_vec.val(i);
+    const double v_val = v_vec.val(i);
+    const double sv_val = sv_vec.val(i);
+
+    const auto params = std::make_tuple(y_val - t0_val, a_val, v_val, w_val,
+                                        sv_val, labstol_wiener5, false);
+
+    dens = internal::estimate_with_err_check<5>(
+        internal::wiener5_helper<internal::FunType::Density>,
+        lerror_bound_dens - LOG_TWO, params, true);
+    ld += dens;
+
+    double new_est_err = dens + lerror_bound - LOG_FOUR;
+
+    // computation of derivative for t and precision check in order to give
+    // the value as deriv_y to edge1 and as -deriv_y to edge5
+    double deriv_y = internal::estimate_with_err_check<5>(
+        internal::wiener5_helper<internal::FunType::GradT>, new_est_err,
+        params);
+
+    // computation of derivatives and precision checks
+    if (!is_constant_all<T_y>::value) {
+      ops_partials.edge1_.partials_[i] = deriv_y;
+    }
+    if (!is_constant_all<T_a>::value) {
+      ops_partials.edge2_.partials_[i] = internal::estimate_with_err_check<5>(
+          internal::wiener5_helper<internal::FunType::GradA>, new_est_err,
+          params);
+    }
+    if (!is_constant_all<T_t0>::value) {
+      ops_partials.edge3_.partials_[i] = -deriv_y;
+    }
+    if (!is_constant_all<T_w>::value) {
+      ops_partials.edge4_.partials_[i] = internal::estimate_with_err_check<5>(
+          internal::wiener5_helper<internal::FunType::GradW>, new_est_err,
+          params);
+    }
+    if (!is_constant_all<T_v>::value) {
+      ops_partials.edge5_.partials_[i]
+          = internal::wiener5_helper<internal::FunType::GradV>(
+              y_val - t0_val, a_val, v_val, w_val, sv_val);
+    }
+    if (!is_constant_all<T_sv>::value) {
+      ops_partials.edge6_.partials_[i]
+          = internal::wiener5_helper<internal::FunType::GradSV>(
+              y_val - t0_val, a_val, v_val, w_val, sv_val);
+    }
```

Comment:
I don't like this style of `weiner5_helper` with an enum. Why can't this code just be inlined here? `weiner5_helper` has a lot of if statements which tbh makes me feel like the calculations don't share that much code